### PR TITLE
Arrays can now be read in every case, even at the root of the document

### LINKED ISSUE #1224: Trying to read a document that is an array results in error
When writing a document that is an array, such as [1, 2, 3] - no error occurs. However upon reading that document a 500 occurs every time. This is a regression as this used to work in previous versions.

### PR DESCRIPTION
**Pull request by EricBorczuk**
**What this PR does**:
Fixes a regression that was recently encountered, where root level array data could be written but could not be read back out.  When reading back out a 500 would occur consistently.

**Which issue(s) this PR fixes**:
Fixes #1224

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated

### CODE REVIEW

File: restapi/src/main/java/io/stargate/web/docsapi/service/JsonConverter.java [hunk 1]
```diff
@@ -50,6 +50,24 @@ public JsonNode convertToJsonDoc(
     return convertToJsonDoc(rows, collector, writeAllPathsAsObjects, numericBooleans, maxDepth);
   }
 
+  /**
```
Comment (tatu-at-datastax) [inline restapi/src/main/java/io/stargate/web/docsapi/service/JsonConverter.java:53]: Really appreciate a good Javadoc here, +1!

**Comment by tatu-at-datastax** [comment]
One beginner question: are root-level arrays expected valid content? I assume so, just wanted to verify as requiring (JSON) Object at root level is something databases could legitimately do.

**Comment by EricBorczuk** [comment]
So I remembered that initially root-level arrays were allowed, and suspected that at some point things regressed, so I looked back in git history; and sure enough there was a point where the code got refactored and the support for it sort of got dropped (and there was no test to cover it)

At any rate, I think it is legitimate to allow root-level docs - search will still work with them (you could search using the array path brackets like `[0].whatever`), and since it's valid JSON I think it is reasonable to allow them as documents!

**Comment by tatu-at-datastax** [comment]
@EricBorczuk this is probably something Lorene could maybe help with once she is back?

File: restapi/src/main/java/io/stargate/web/docsapi/service/JsonConverter.java [hunk 2]
```diff
@@ -121,6 +139,11 @@ private JsonNode convertToJsonDoc(
         if (isArray && !writeAllPathsAsObjects) {
           int index = Integer.parseInt(p.substring(1, p.length() - 1));
 
+          boolean shouldBeArray = isArray && !ref.isArray();
+          if (i == 0 && shouldBeArray) {
```
Comment (tomekl007) [inline restapi/src/main/java/io/stargate/web/docsapi/service/JsonConverter.java:143]: why onlt if `i == 0`?

Comment (ivansenic) [reply restapi/src/main/java/io/stargate/web/docsapi/service/JsonConverter.java:143]: Good point, seems that we would end up in the cast exception if `i` is not `0` and it should be array?

File: testing/src/main/java/io/stargate/it/http/docsapi/BaseDocumentApiV2Test.java [hunk 3]
```diff
@@ -535,6 +535,15 @@ public void testPutNullsAndEmpties() throws IOException {
         .isEqualTo(wrapResponse(OBJECT_MAPPER.createArrayNode(), "4", null));
   }
 
+  @Test
+  public void testPutSimpleArray() throws IOException {
```
Comment (ivansenic) [inline testing/src/main/java/io/stargate/it/http/docsapi/BaseDocumentApiV2Test.java:539]: should we consider adding other edge cases here? 

* I guess normal string is a valid JSON, should we test with string, blank string empty string?
* same with numbers, booleans
* empty documents (I guess this is not allowed at the moment)

File: restapi/src/main/java/io/stargate/web/docsapi/service/JsonConverter.java (hunk 2 above)
Comment (tatu-at-datastax) [reply restapi/src/main/java/io/stargate/web/docsapi/service/JsonConverter.java:143]: If entries are sorted first, wouldn't we always get `[0]` entry first? Unless that was filtered I guess?
Then again there should be no harm in checking it for further entries either way.

Comment (EricBorczuk) [reply restapi/src/main/java/io/stargate/web/docsapi/service/JsonConverter.java:143]: So I took a look at this code, and yes - for completeness there should be a clause to deal with converting an object to an array when it is not at the root - I added that.

However, because of how row deletions occur before inserts in the API, there isn't a real situation at the moment that causes an object to have to be converted to an array - which is why this issue has never been hit. At any rate, I added a test that should cover this behavior, as well, in case something changes in the future to how we delete data before writing.

File: testing/src/main/java/io/stargate/it/http/docsapi/BaseDocumentApiV2Test.java (hunk 3 above)
Comment (EricBorczuk) [reply testing/src/main/java/io/stargate/it/http/docsapi/BaseDocumentApiV2Test.java:539]: These are all disallowed, and tested by [testInvalidPuts](https://github.com/stargate/stargate/blob/bc9b7f509fb498044353e5d6ff63e39e2d720936/testing/src/main/java/io/stargate/it/http/docsapi/BaseDocumentApiV2Test.java#L665)

File: restapi/src/main/java/io/stargate/web/docsapi/service/JsonConverter.java (hunk 2 above)
Comment (tatu-at-datastax) [reply restapi/src/main/java/io/stargate/web/docsapi/service/JsonConverter.java:143]: ahhh. My comment was in error; I did not pay attention and was thinking of 0 as the first element, as opposed to the first path segment.

**Review by ivansenic — APPROVED** [review]
Seems good..
